src/main/java/org/htwkvisu/gui/MapCanvas.java

```diff
@@ -116,22 +139,50 @@ public Point2D getCenter() {
         return mapCenter;
     }
 
+    public int getSamplingPixelDensity() {
+        return samplingPixelDensity;
+    }
+
+    public void setSamplingPixelDensity(int samplingPixelDensity) {
+        if (samplingPixelDensity > 0) {
+            this.samplingPixelDensity = samplingPixelDensity;
+        }
+    }
+
     /**
-     * Set the maps scale.
+     * Returns coordinates for samples where to calculate scoring values.
+     * They are positioned in a XY-grid with pixelDensity pixels distance between each sample.
      *
-     * @param scale of map
+     * @param pixelDensity Distance between samples in pixels
+     * @return Sample positions in a grid
      */
-    public void setScale(double scale) {
-        if (scale < ZOOM_MAX) {
-            if (scale > ZOOM_MIN) {
-                this.scale = scale;
-            } else {
-                this.scale = ZOOM_MIN;
+    public ArrayList<ArrayList<Point2D>> getSampleCoordPoints(float pixelDensity) {
+        ArrayList<ArrayList<Point2D>> matrix = new ArrayList<>();
+
+        double coordsDistance = getCoordDistanceFromPixelDistance(pixelDensity);
+
+        // create sample coordinates for the currently drawn map area
+        // important: to get scoring values for all shown pixles, the code will create samples around the canvas as well
+
+        // longitude
+        for (double x = getLeftBottomCorner().getX() - coordsDistance; x < getLeftTopCorner().getX() + coordsDistance; x = x + coordsDistance) {
```

Comment:
Just in case you want to do this with only x or y - use DoubleStream:
`DoubleStream.iterate(getLeftBottomCorner().getX(), x -> x + coordsDistance)
                .limit((new Double(getLeftTopCorner().getX()+coordsDistance)).longValue())
                .boxed().collect(Collectors.toList());`
I don't know if its better to use this here too, its your decision.